Tests/Editor/Geometry/DuplicateFacesTest.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UObject = UnityEngine.Object;
+using NUnit.Framework;
+using UnityEngine.ProBuilder;
+using UnityEditor.ProBuilder;
+using UnityEditor;
+using UnityEngine.TestTools;
+
+
+namespace UnityEngine.ProBuilder.EditorTests.Geometry
+{
+    class DuplicateFacesTest
+    {
+        ProBuilderMesh[] selectables;
+
+        [SetUp]
+        public void Setup()
+        {
+            ProBuilderMesh shape1 = ShapeGenerator.CreateShape(ShapeType.Cube);
+            shape1.transform.position = Vector3.zero - shape1.GetComponent<MeshRenderer>().bounds.center;
+
+            selectables = new ProBuilderMesh[]
+            {
+                shape1
+            };
+        }
+
+        [TearDown]
+        public void Cleanup()
+        {
+            for (int i = 0; i < selectables.Length; i++)
+            {
+                UObject.DestroyImmediate(selectables[i].gameObject);
+            }
+        }
+
+        [Test]
+        public void DuplicateFaces_ToObject()
+        {
+            //Make first faces selected
+            ProBuilderMesh mesh = selectables[0];
+            Assert.IsNotNull(mesh.faces);
```

Comment:
If this is just part of the test setup and not what you're actually testing, this should be `Assume.That(mesh.faces, Is.NotNull)`.

Reply:
Done!